### PR TITLE
refactor: move `UnsupportedCompiler` to errors

### PR DESCRIPTION
**Pull request by jakmeier**
This is an actual error, not a user abort. It should be treated as such,
therefore it belongs to `VMRunnerError` and not `CompilationError`
which is part of the use abort type aka `FunctionCallError`.

`CompilationError` implements `BorshSerialize` and is used in
`FunctionCallErrorSer`. Removing `UnsupportedCompiler` is okay because:
1) The error only gets thrown from unused`precompile`. It is is dead
code in production.
2) It is the last variant in the enum and hence does not affect the
borsh serialization format of other variants.

cc #7818

### CODE REVIEW

File: runtime/near-vm-errors/src/lib.rs [hunk 1]
```diff
@@ -25,6 +25,9 @@ pub enum VMRunnerError {
     WasmUnknownError {
         debug_message: String,
     },
+    UnsupportedCompiler {
```
Comment (matklad) [inline runtime/near-vm-errors/src/lib.rs:28]: While we are at it, add comment explaining when we emit this error

**Comment by matklad** [comment]
>The error only gets thrown from unusedprecompile. It is is dead
code in production.

Ah, that means I didn't quite fully remove precompilation, will fix this as a followup

**Review by matklad — APPROVED** [review]

File: runtime/near-vm-errors/src/lib.rs (hunk 1 above)
Comment (matklad) [reply runtime/near-vm-errors/src/lib.rs:28]: Scratch that, I think I'd just kill this variant

Comment (jakmeier) [reply runtime/near-vm-errors/src/lib.rs:32]: too late now^^ but yeah if we delete the precompile code next, then this can be deleted as well

Comment (matklad) [reply runtime/near-vm-errors/src/lib.rs:32]: https://github.com/near/nearcore/pull/7851

:weary: one does never "just" do a thing

**Pull request merged**